backend/lib/endpoints/posts.js

```diff
@@ -222,13 +223,35 @@ async function routes(app) {
         throw app.httpErrors.forbidden();
       }
 
+      // Author defaults to user unless organizationId set
+      let author = user;
+      const { organizationId } = postProps;
+      if (organizationId) {
+        const [orgErr, org] = await app.to(User.findById(organizationId));
+        if (orgErr) {
+          req.log.error(userErr, "Failed retrieving organization");
+          throw app.httpErrors.internalServerError();
+        } else if (org === null) {
+          req.log.error(userErr, "Organization does not exist");
+          throw app.httpErrors.forbidden();
+        } else if (org.ownerId.toString() !== userId.toString()) {
```

Comment:
Instead of converting both to string we could use the `.equals()` from the ObjectId

Reply:
so from `org.ownerId.toString() !== userId.toString()` to -> `org.ownerId.equals(userId)`? I also think that would be better. I guess we can revisit later.

Reply:
Yep, that's correct